### PR TITLE
Refactor editorconfig-conf-mode

### PR DESCRIPTION
**Pull request by xuhdev**
@10sr Can you review this PR? Thanks!

### CODE REVIEW

File: editorconfig.el [hunk 1]
```diff
@@ -440,31 +440,50 @@ visiting files or changing major modes if the major mode is not listed in
...
+            "space"
+            "tab"
+            "latin1"
+            "utf-8"
+            "utf-8-bom"
+            "utf-16be"
+            "utf-16le"))
+        (font-lock-value
+          '(("^[ \t]*\\[\\(.+?\\)\\]" 1 'font-lock-type-face)
+            ("^[ \t]*\\(.+?\\)[ \t]*[=:]" 1 'font-lock-variable-name-face))))
```
Comment (10sr) [inline editorconfig.el]: Sorry this is my mistake, but I think these two single quotes (`'`) before face names are not required.

File: editorconfig.el [hunk 2]
```diff
@@ -440,31 +440,50 @@ visiting files or changing major modes if the major mode is not listed in
...
+
+    ;; Highlight all key values
+    (dolist (value-property value-property-list)
+      (add-to-list
+        'font-lock-value
+        `(,(format "[=:][ \t]*\\(%s\\)\\([ \t]\\|$\\)" value-property)
+          1 font-lock-builtin-face)))
```
Comment (10sr) [inline editorconfig.el]: Typo for `font-lock-constant-face` ? :bow:

File: editorconfig.el [hunk 3]
```diff
@@ -440,31 +440,50 @@ visiting files or changing major modes if the major mode is not listed in
...
+        (font-lock-value
+          '(("^[ \t]*\\[\\(.+?\\)\\]" 1 'font-lock-type-face)
+            ("^[ \t]*\\(.+?\\)[ \t]*[=:]" 1 'font-lock-variable-name-face))))
+
+    ;; Highlight all key values
+    (dolist (key-value key-value-list)
+      (add-to-list
+        'font-lock-value
+        `(,(format "[=:][ \t]*\\(%s\\)\\([ \t]\\|$\\)" key-value)
+          1 font-lock-builtin-face)))
```
Comment (10sr) [inline editorconfig.el]: Typo for `font-lock-constant-face` ? :bow:

**Comment by 10sr** [comment]
Looks pretty nice for the rest ! 
Thanks !!!

**Comment by xuhdev** [comment]
All fixed, thanks!

**Comment by 10sr** [comment]
:+1:

**Pull request merged**